react/features/stream-effects/presenter/JitsiStreamPresenterEffect.js

```diff
@@ -0,0 +1,164 @@
+        const { height, width, frameRate } = firstVideoTrack.getSettings()
+            ?? firstVideoTrack.getConstraints();
+
+        this._canvas = document.createElement('canvas');
+
+        // Workaround for FF issue
+        // https://bugzilla.mozilla.org/show_bug.cgi?id=1388974
+        this._ctx = this._canvas.getContext('2d');
+        document.body.appendChild(this._canvas);
+        this._desktopElement = document.createElement('video');
+        this._videoElement = document.createElement('video');
+        document.body.appendChild(videoDiv);
+        videoDiv.appendChild(this._videoElement);
+        videoDiv.appendChild(this._desktopElement);
+
+        // Set the video element properties
+        this._frameRate = frameRate;
+        this._videoElement.width = width;
+        this._videoElement.height = height;
+        this._videoElement.autoplay = true;
+        this._videoElement.srcObject = videoStream;
+
+        // set the style attribute of the div to make it invisible
+        videoDiv.style.display = 'none';
+
+        // Bind event handler so it is only bound once for every instance.
+        this._onVideoFrameTimer = this._onVideoFrameTimer.bind(this);
+        this._videoFrameTimerWorker = new Worker(timerWorkerScript);
+        this._videoFrameTimerWorker.onmessage = this._onVideoFrameTimer;
+    }
+
+    /**
+     * EventHandler onmessage for the videoFrameTimerWorker WebWorker.
+     *
+     * @private
+     * @param {EventHandler} response - The onmessage EventHandler parameter.
+     * @returns {void}
+     */
+    _onVideoFrameTimer(response) {
+        if (response.data.id === INTERVAL_TIMEOUT) {
+            this._renderVideo();
+        }
+    }
+
+    /**
+     * Loop function to render the video frame input and draw presenter effect.
+     *
+     * @private
+     * @returns {void}
+     */
+    _renderVideo() {
+        this._ctx.drawImage(this._desktopElement, 0, 0, this._desktopElement.width,
+            this._desktopElement.height);
+        this._ctx.drawImage(this._videoElement, this._videoCordinates.x,
+            this._videoCordinates.y, this._videoCordinates.width, this._videoCordinates.height);
+
+        // draw a border around the video element.
+        this._ctx.beginPath();
+        this._ctx.lineWidth = '2';
+        this._ctx.strokeStyle = '#A9A9A9'; // dark grey
+        this._ctx.rect(this._videoCordinates.x, this._videoCordinates.y,
+            this._videoCordinates.width, this._videoCordinates.height);
+        this._ctx.stroke();
+    }
+
+    /**
+     * Checks if the local track supports this effect.
+     *
+     * @param {JitsiLocalTrack} jitsiLocalTrack - Track to apply effect.
+     * @returns {boolean} - Returns true if this effect can run on the
+     * specified track, false otherwise.
+     */
+    isEnabled(jitsiLocalTrack) {
+        return jitsiLocalTrack.isVideoTrack()
+            && jitsiLocalTrack.videoType === 'desktop';
```

Comment:
doesn't this fit in a single line?

Reply:
Are we not sticking to 80 characters per line limit ?